src/lib/SpotlightTour.provider.tsx

```diff
@@ -3,40 +3,88 @@ import { ColorValue, LayoutRectangle } from "react-native";
 
 import { ChildFn, isChildFunction, isPromise } from "../helpers/common";
 
-import { SpotlightTour, SpotlightTourContext, SpotlightTourCtx, TourStep } from "./SpotlightTour.context";
+import { Position, SpotlightTour, SpotlightTourContext, SpotlightTourCtx, TourStep } from "./SpotlightTour.context";
 import { TourOverlay, TourOverlayRef } from "./tour-overlay/TourOverlay.component";
 
+export interface OSConfig<T> {
+  android: T;
+  ios: T;
+}
+
 interface SpotlightTourProviderProps {
+  /**
+   * The children to render in the provider. It accepts either a React
+   * component, or a function that returns a React component. When the child is
+   * a funtion, the `SpotlightTour` context can be accessed from the first
+   * argument.
+   */
   children: React.ReactNode | ChildFn<SpotlightTour>;
+  /**
+   * The color of the overlay of the tour.
+   *
+   * @default black
+   */
   overlayColor?: ColorValue;
-  overlayOpacity?: number | string;
+  /**
+   * The opacity applied to the overlay of the tour (between 0 to 1).
+   *
+   * @default 0.45
+   */
+  overlayOpacity?: number;
+  /**
+   * An array of `TourStep` objects that define each step of the tour.
+   */
   steps: TourStep[];
+  /**
+   * Define if the animations in the tour should use the native driver or not.
+   * A boolean can be used to apply the same value to both Android and iOS, or
+   * an object with `android` and `ios` keys can be used to define a value for
+   * each OS.
+   *
+   * @default false
+   */
+  nativeDriver?: boolean | OSConfig<boolean>;
 }
 
+const ZERO_SPOT: LayoutRectangle = {
+  height: 0,
+  width: 0,
+  x: 0,
+  y: 0
+};
+
 export const SpotlightTourProvider = React.forwardRef<SpotlightTour, SpotlightTourProviderProps>((props, ref) => {
-  const { children, overlayColor, overlayOpacity, steps } = props;
+  const {
+    children,
+    overlayColor = "black",
+    overlayOpacity = 0.45,
+    steps,
+    nativeDriver = false
+  } = props;
 
   const [current, setCurrent] = useState<number>();
-  const [spot, setSpot] = useState<LayoutRectangle>();
+  const [spot, setSpot] = useState(ZERO_SPOT);
 
-  const overlayRef = useRef<TourOverlayRef>(null);
+  const overlay = useRef<TourOverlayRef>({
+    hideTooltip: () => Promise.resolve({ finished: false })
+  });
 
   const renderStep = useCallback((index: number) => {
```

Comment:
All of the `renderStep(n)` calls do not await the result of this promise, is this expected or should we care about the non `Promise.resolve()` scenarios?

Reply:
Yes, those are expected because the callbacks where they are called are rendering effects, so they return `void`. Waiting for the promise will cause them to return `Promise<void>` instead, and there's no way to ensure that the render is done once the promise resolves.

But a good point there is that we don't really need to return the promise in `renderStep` either 🙂